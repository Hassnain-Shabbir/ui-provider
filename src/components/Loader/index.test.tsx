import React from "react";
import { render } from "@testing-library/react";
import {
  createRenderer as createShallowRenderer,
  ShallowRenderer,
} from "react-test-renderer/shallow";

import Component, { Props } from "./index";

describe("snapshot", () => {
  let renderer: ShallowRenderer;

  beforeEach(() => {
    renderer = createShallowRenderer();
  });
  afterEach(() => {
    renderer.unmount();
  });

  it("mount only", () => {
    renderer.render(<Component />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
  it("fullPage = true", () => {
    const props: Props = {
      fullPage: true,
    };
    const component = <Component {...props} />;

    renderer.render(component);
    expect(renderer.getRenderOutput()).toMatchSnapshot();

    const { container } = render(component);
    const obj = container.querySelector(".page-loader-wrapper");
    expect(obj).toBeDefined();
    expect(obj).not.toBeNull();
  });
  it("absolute = true", () => {
    const props: Props = {
      absolute: true,
    };
    const component = <Component {...props} />;
    renderer.render(component);
    expect(renderer.getRenderOutput()).toMatchSnapshot();

    const { container } = render(component);
    const obj = container.querySelector(".absolute");
    expect(obj).toBeDefined();
    expect(obj).not.toBeNull();
  });
});
