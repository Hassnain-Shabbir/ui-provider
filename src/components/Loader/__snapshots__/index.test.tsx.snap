// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot absolute = true 1`] = `
<div
  className="loader loader-wrapper absolute"
>
  <LoaderIcon
    className="icon icon"
  />
</div>
`;

exports[`snapshot fullPage = true 1`] = `
<div
  className="loader page-loader-wrapper"
>
  <LoaderIcon
    className="icon icon full-page"
  />
</div>
`;

exports[`snapshot mount only 1`] = `
<div
  className="loader loader-wrapper"
>
  <LoaderIcon
    className="icon icon"
  />
</div>
`;
